torch/lib/c10d/ProcessGroupNCCL.cpp

```diff
@@ -624,14 +624,10 @@ std::vector<at::Tensor> flatten_for_scatter_gather(
     }
 
     for (const auto& t : tensor_lists[i]) {
-      if (t.sizes() != other[i].sizes()) {
+      if (t.numel() != other[i].numel()) {
         throw std::runtime_error(
             "All tensor operands to scatter/gather must have the same size");
```

Comment:
Suggestion, which you can take or leave: change "the same" to "identical"...not because it's better phrasing, but so that when people ping us with an error message, we know immediately they're using this version of the check.